docs/content/documentation/systems/Kernels/framework/MaterialDerivativeRankFourTestKernel.md

```diff
@@ -0,0 +1,14 @@
+# MaterialDerivativeRankFourTestKernel
+!syntax description /Kernels/MaterialDerivativeRankFourTestKernel
+
+This kernel puts a seleted tensorial (`RankFourTensor` type) material property (`material_property`) in the residual vector
+and assembles the Jacobian using the derivatives of the material property as provided by the
+[DerivativeMaterialInterface](/DerivativeMaterialInterface.md).
```

Comment:
The name implies that this is for testing purposes only, but I'd explicitly say that in the documentation. Same for the RankTwoTensor version.

Reply:
Yes, good suggestion. Any objections to me doing this in the pending follow up PR?

Reply:
Actually a comment to that effect is already in the class description:

```
  params.addClassDescription(
      "Class used for testing derivatives of a rank four tensor material property.");
```